node/src/accountant/big_int_db_processor.rs

```diff
@@ -0,0 +1,1613 @@
+                row.get::<usize, i64>(1).expectv("low_bytes"),
+            ))
+        })
+        .expect("select query failed")
+        .flatten()
+        .sum()
+}
+
+macro_rules! create_big_int_sqlite_fns {
+    ($conn: expr, $flags: expr, $($sqlite_fn_name: expr),+; $($intern_fn_name: ident),+) => {
+        $($conn.create_scalar_function::<_, i64>($sqlite_fn_name, 2, $flags, move |ctx| {
+            Ok(BigIntDivider::$intern_fn_name(common_arg_distillation(
+                ctx,
+                $sqlite_fn_name,
+            )?))
+        })?;)+
+    }
+}
+
+macro_rules! parse_fn_creation_args {
+    ($ctx: expr, $fn_name: expr, $($idx: expr),+; $($parser: ident),+; $($err_msg: literal),+) => {
+        ($($ctx.get_raw($idx)
+            .$parser()
+            .map_err(|_| invalid_input_error($fn_name, format!($err_msg, $ctx.get_raw($idx))))?
+        ),+)
+    };
+}
+
+pub struct BigIntDivider {}
+
+impl BigIntDivider {
+    pub fn deconstruct(num: i128) -> (i64, i64) {
+        (
+            Self::deconstruct_high_bytes(num),
+            Self::deconstruct_low_bytes(num),
+        )
+    }
+
+    fn deconstruct_high_bytes(num: i128) -> i64 {
+        let high_bytes = (num >> 63) as i64;
+        if num.is_positive() && (high_bytes.unsigned_abs() & 0xC000000000000000u64) > 0 {
+            panic!("Too big positive integer to be divided: {:#X}", num)
+        }
+        if num < -0x40000000000000000000000000000000 {
+            panic!("Too big negative integer to be divided: -{:#X}", num)
+        }
+        high_bytes
+    }
+
+    fn deconstruct_low_bytes(num: i128) -> i64 {
+        (num & 0x7FFFFFFFFFFFFFFFi128) as i64
+    }
+
+    pub fn reconstitute(high_bytes: i64, low_bytes: i64) -> i128 {
+        let low_bytes = low_bytes as i128;
+        let high_bytes = high_bytes as i128;
+        (high_bytes << 63) | low_bytes
+    }
+
+    pub fn register_deconstruct_for_sqlite_connection(conn: &Connection) -> rusqlite::Result<()> {
+        Self::guts_of_register_deconstruct(conn, "biginthigh", "bigintlow")
+    }
+
+    fn guts_of_register_deconstruct(
+        conn: &Connection,
+        fn_name_1: &'static str,
+        fn_name_2: &'static str,
+    ) -> rusqlite::Result<()> {
+        fn invalid_input_error(fn_name: &str, message: String) -> Error {
+            UserFunctionError(Box::new(InvalidInputValue(fn_name.to_string(), message)))
+        }
+        fn negativity_check_and_final_composition(
+            fn_name: &str,
+            tuple: (i64, f64),
+        ) -> rusqlite::Result<i128> {
+            let (point_to_decrease_from_gwei, decrease_wei) = tuple;
+            if decrease_wei.is_sign_negative() {
+                Ok(point_to_decrease_from_gwei as i128 * WEIS_OF_GWEI + decrease_wei as i128)
+            } else {
+                Err(invalid_input_error(
+                    fn_name,
+                    format!(
+                        "None negative slope, while designed only for use with negative one: {}",
```

Comment:
Error message should say something about young debts, and make it clear that the developer should have written SQL that prevented the young debts from getting this far. (In other words, this isn't a user or data error.)